src/hooks/useV2Pairs.ts

```diff
@@ -22,82 +23,91 @@ export enum PairState {
 }
 
 export function useV2Pairs(currencies: [Currency | undefined, Currency | undefined][]): [PairState, Pair | null][] {
-  const [reserves, setReserves] = useState<
-    ({ reserve0: string; reserve1: string; poolId: number; totalSupply: string } | number | undefined)[]
-  >([]);
-
   const tokens = useMemo(() => {
     return currencies.map(([currencyA, currencyB]) => [currencyA?.wrapped, currencyB?.wrapped]);
   }, [currencies]);
 
-  // const lastBlockNumber = useBlockNumber();
+  const [pairs, setPairs] = useState<[PairState, Pair | null][]>(Array(tokens.length).fill([PairState.LOADING, null]));
+
+  const lastBlockNumber = useBlockNumber();
 
   useEffect(() => {
-    setReserves(Array(tokens.length).fill(PairState.LOADING));
+    setPairs(Array(tokens.length).fill([PairState.LOADING, null]));
+  }, [tokens]);
 
+  useEffect(() => {
     const fetchReserves = async () => {
       try {
         const result = await Promise.all(
-          tokens.map(async ([tokenA, tokenB]) => {
-            if (tokenA && tokenB && tokenA.chainId === tokenB.chainId && !tokenA.equals(tokenB)) {
-              let stats;
-              let poolId;
-
-              try {
-                poolId = parseInt(await bnJs.Dex.getPoolId(tokenA.address, tokenB.address), 16);
-                if (poolId === 0) return undefined;
-                stats = await bnJs.Dex.getPoolStats(poolId);
-              } catch (err) {
-                return undefined;
+          tokens.map(
+            async ([tokenA, tokenB]): Promise<[PairState, Pair | null]> => {
+              if (tokenA && tokenB && tokenA.chainId === tokenB.chainId && !tokenA.equals(tokenB)) {
+                let stats;
+                let poolId;
+
+                try {
+                  poolId = parseInt(await bnJs.Dex.getPoolId(tokenA.address, tokenB.address), 16);
+                  if (poolId === 0) return [PairState.NOT_EXISTS, null];
+                  stats = await bnJs.Dex.getPoolStats(poolId);
+                } catch (err) {
+                  return [PairState.NOT_EXISTS, null];
+                }
+
+                const baseReserve = new BigNumber(stats['base'], 16).toFixed();
+                const quoteReserve = new BigNumber(stats['quote'], 16).toFixed();
+                const totalSupply = new BigNumber(stats['total_supply'], 16).toFixed();
+
+                if (stats['base_token'] === tokenA.address) {
```

Comment:
Can we refactor this?

`const [reserveA, reserveB] = (stats['base_token'] === tokenA.address) ? [baseReserve, quoteReserve] : [quoteReserve, baseReserve];`